Checks/PWR079/README.md

```diff
@@ -0,0 +1,336 @@
+# PWR079: Avoid undefined behavior due to uninitialized variables
+
+### Issue
+
+Accessing an uninitialized variable can lead to undefined behavior due to its
+indeterminate value.
+
+### Actions
+
+Always initialize variables before using them, helping ensure deterministic
+behavior and prevent bugs in the code.
+
+### Relevance
+
+Some programming languages automatically initialize variables to default
+values, but Fortran, C, and C++ often do not. Consequently, variables left
+uninitialized contain unpredictable data until they are explicitly set by the
+programmer:
+
+- **Fortran:** Reading any variable that has not been explicitly initialized
+  results in undefined behavior.
+
+- **C:** Automatic (i.e., declared within functions) variables are not
+  initialized by default. However, `static`, `thread_local`, and file-scope
+  variables are zero-initialized (e.g., scalar values are set to `0`).
+
+- **C++:** Similar to C, with additional rules for default initialization of
+  class-like objects and array elements in certain contexts.
+
+Since uninitialized variables may contain any arbitrary values, reading and
+using them can lead to undefined behavior, potentially causing incorrect
+results, crashes, or other unintended outcomes. Compilers are not required to
+warn about these issues and, even if they do, they typically still allow the
+code to compile and run.
+
+Some compilers may appear to "help" by zero-initializing variables under
+certain conditions (e.g., specific compilation flags). While this can make
+technically incorrect code run as originally intended, relying on such
+incidental behavior creates a false sense of security and masks underlying
+logical errors. Ultimately, this can vanish under different compilation
+settings and can vary between compilers.
+
+Lastly, while some compilers provide options to automatically initialize
+certain data types (e.g., `gfortran`'s `-finit-integer=<value>` or `gcc`'s
+`-ftrivial-auto-var-init=<value>`), these features reduce portability to other
+development environments and hide problems in the code rather than addressing
+them.
+
+### Code examples
+
+#### C
+
+Consider the following code, which aims to sum the elements of an array:
+
+```c
+// example_array.c
+#include <stdio.h>
+
+__attribute__((pure)) double sum_array(double *array, int size) {
```

Comment:
Nit:
```suggestion
__attribute__((pure)) double sum_array(double *array, size_t size) {
```

Reply:
I usually opt for `int` to prevent including a header, but since `stdio.h` is already used within this code example, fine by me! :)